packages/es-components/src/components/controls/switch/Switch.js

```diff
@@ -0,0 +1,226 @@
+  justify-content: ${props => props.direction === 'left' ? 'flex-end' : 'normal'};
+  cursor: pointer;
+  display: flex;
+  position: relative:
+  padding: 5px 0;
+  transition: all .25s linear;
+
+  &:hover {
+    span {
+      &::before {
+        box-shadow: inset 0 3px 3px rgba(0,0,0,0.4);
+      }
+    }
+
+    input[type="checkbox"]:checked ~ span::before {
+      box-shadow: inset 0 3px 3px rgba(0,0,0,0.7);
+    }
+  }
+`;
+
+const SwitchInput = styled.input`
+  overflow: visible;
+  margin: 0;
+  font-family: inherit;
+  font-size: inherit;
+  line-height: inherit;
+  position: absolute;
+  clip: rect(0,0,0,0);
+  pointer-events: none;
+  box-sizing: border-box;
+  padding: 0;
+
+  &:focus ~ span::after {
+    box-shadow: 0 0 3px 3px #83bffc;
+  }
+
+  &:checked ~ span {
+    &::before {
+      background: ${props => Type(props.styleType, false)};
+    }
+
+    &::after {
+      background: #f4f4f4;
+      left: 28px;
+    }
+  }
+`;
+
+const SwitchLabelText = styled.span`
+  padding: ${props => Padding[props.direction]};
+  flex-grow: 1;
+  color: ${props => props.type === 'primary' ? '#000' : Type(props.type, false)};
+`;
+
+const SwitchCheck = styled.span`
+  order: -2;
+  position: relative;
+  width: 64px;
+
+  &::before {
+    background: ${props => Type(props.type, true)};
+    border-radius: 11px;
+    content: "";
+    display: block;
+    height: 25px;
+    left: 3px;
+    outline: none;
+    position: absolute;
+    top: .2rem;
+    width: 54px;
+    z-index: 5;
+    transition: background-color .25s linear,border .25s linear,box-shadow .25s linear;
+  }
+  
+  &::after {
+    background: #f4f4f4;
+    box-shadow: 0 2px 6px #444;
+    display: block;
+    position: absolute;
+    content: "";
+    border-radius: 100%;
+    height: 30px;
+    width: 30px;
+    top: 0;
+    left: 0;
+    transform: all .18s linear;
+    z-index: 7;
+  }
+`;
+
+const SwitchOffText = styled.span`
+  order: ${props => props.direction === 'left' || props.direction === 'top' ? '-1' : '-3'};
+  padding: ${props => props.offText === '' ? '0' : '0 1rem 0 1rem'};;
+  vertical-align: middle;
+`
+
+const SwitchOnText = styled.span`
+  order: ${props => props.direction === 'left' || props.direction === 'top' ? '-3' : '-1'};
+  padding: ${props => props.onText === '' ? '0' : '0 1rem 0 0.75rem'};
+  vertical-align: middle;
+`
+
+const disabledProp = {
+  cursor: 'not-allowed'
+}
+
+function Switch({
+  type,
+  label,
+  direction,
+  ariaLabel,
+  onText,
+  offText,
+  checked,
+  onChange,
+  disabled,
+  ...switchProps
+}) {
+  const [isToggled, setIsToggled] = useState(checked);
+
+  function handleToggle(event) {
+    setIsToggled(!isToggled);
+    onChange(event);
+  }
+
+  return (
+    <SwitchBase
+      type={type}
+      {...switchProps}
+    >
+      <SwitchLabel direction={direction} onChange={handleToggle} style={disabled ? disabledProp : {}}>
```

Comment:
You can simply pass the `disabled` prop into `SwitchLabel` (probably using a slightly different name like `isDisabled` to avoid confusion with the actual `disabled` dom attribute) and use a ternary to define the cursor.
```
  <SwitchInput isDisabled={disabled} ... />
```
and within `SwitchLabel`:
```
  cursor: ${props => props.isDisabled ? 'not-allowed' : 'pointer'};
```

Other than that, this looks great!

Reply:
That makes sense, also added some other disabled conditionals in there.